databricks/koalas/utils/accessors.py

```diff
@@ -0,0 +1,177 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+"""
+Utility features. The features here are specific to Koalas.
+"""
+from collections import OrderedDict
+from typing import Tuple, Union, TYPE_CHECKING
+
+from databricks.koalas.internal import InternalFrame, SPARK_INDEX_NAME_FORMAT
+from databricks.koalas.utils import name_like_string, scol_for
+
+if TYPE_CHECKING:
+    from databricks.koalas.frame import DataFrame
+
+
+class UtilsFrameMethods(object):
+    """ Utility features for DataFrame. Usually, The features here are specific to Koalas. """
+
+    def __init__(self, frame: "DataFrame"):
+        self._kdf = frame
+
+    def attach_sequence_column(self, column: Union[str, Tuple[str, ...]]) -> "DataFrame":
+        """
+        Attach a column of a sequence that increases one by one.
+
+        .. note:: this uses Spark's Window without specifying partition specification.
+            This leads to move all data into single partition in single machine and
+            could cause serious performance degradation.
+            Avoid this method against very large dataset.
+
+        Parameters
+        ----------
+        column : string or list of string
+            The column name to be attached.
+
+        Returns
+        -------
+        DataFrame
+            The DataFrame attached the column.
+
+        Examples
+        --------
+        >>> df = ks.DataFrame({"x": ['a', 'b', 'c']})
+        >>> df.utils.attach_sequence_column("sequence")
+           x  sequence
+        0  a         0
+        1  b         1
+        2  c         2
+        """
+        return self._attach_default_index_like_column(column, InternalFrame.attach_sequence_column)
+
+    def attach_distributed_sequence_column(
+        self, column: Union[str, Tuple[str, ...]]
+    ) -> "DataFrame":
+        """
+        Attach a column of a sequence that increases one by one, by group-by and group-map approach
+        in a distributed manner.
+
+        Parameters
+        ----------
+        column : string or list of string
+            The column name to be attached.
+
+        Returns
+        -------
+        DataFrame
+            The DataFrame attached the column.
+
+        Examples
+        --------
+        >>> df = ks.DataFrame({"x": ['a', 'b', 'c']})
+        >>> df.utils.attach_distributed_sequence_column("distributed_sequence").sort_index()
+           x  distributed_sequence
+        0  a                     0
+        1  b                     1
+        2  c                     2
+        """
+        return self._attach_default_index_like_column(
+            column, InternalFrame.attach_distributed_sequence_column
+        )
+
+    def attach_distributed_column(self, column: Union[str, Tuple[str, ...]]) -> "DataFrame":
```

Comment:
What do you think about `column_name` rather `column` to make users understand this parameter more clearly?

Reply:
Usually pandas doesn't use `column_name` or `column_label`, but `column` or `columns`. I think we should follow the convention.

Reply:
Ah, I agree If pandas is doing so :D